File: r/visualisation.R
library(readr)
library(ggplot2)

# Load dataset
file_path <- 'D:\\Hello World\\diabetes\\diabetes.csv'
df <- read_csv(file_path)

# Histogram
p1 <- ggplot(df, aes(x = Glucose)) +
  geom_histogram(bins = 20) +
  ggtitle('Distribution of Glucose Levels')

print(p1)
ggsave("D:\\Hello World\\diabetes\\outputs\\histogramR.png")

# Boxplot
p2 <- ggplot(df, aes(x = factor(Outcome), y = Age)) +
  geom_boxplot() +
  xlab('Outcome') +
  ylab('Age') +
  ggtitle('Age Distribution by Diabetes Outcome')

print(p2)
ggsave("D:\\Hello World\\diabetes\\outputs\\boxplotR.png")

File: python/visualisation.py
import pandas as pd
import matplotlib.pyplot as plt

# Load data
file_path = 'diabetes.csv'
df = pd.read_csv(file_path)

# Histogram of glucose levels
plt.hist(df['Glucose'])
plt.xlabel('Glucose Level')
plt.ylabel('Frequency')
plt.title('Distribution of Glucose Levels')
plt.savefig('outputs/glucose_histogram.png')
plt.close()

# Boxplot by diabetes outcome
plt.boxplot([
    df[df['Outcome'] == 0]['Age'],
    df[df['Outcome'] == 1]['Age']
])

plt.xticks([1, 2], ['Non-Diabetic', 'Diabetic'])
plt.ylabel('Age')
plt.title('Age Distribution by Diabetes Outcome')
plt.savefig('outputs/age_boxplot.png')
